### PR TITLE
fix: detect JetBrains running on local ipv6

### PR DESCRIPTION
**Pull request by code-asher**
Minus the modified unit test, for now.  Will put up a fixed test in a separate PR!

### CODE REVIEW

File: agent/agentssh/portinspection_supported.go [hunk 1]
```diff
@@ -11,24 +12,33 @@ import (
...
 	}
-	if len(tabs) == 0 {
-		return "", nil
+	tabs, err := netstat.TCPSocks(acceptFn)
+	tabs6, err6 := netstat.TCP6Socks(acceptFn)
+
+	// Only return the error if the other method found nothing.
+	if (err != nil && len(tabs6) == 0) || (err6 != nil && len(tabs) == 0) {
+		return "", xerrors.Errorf("inspect port %d: %w", port, errors.Join(err, err6))
 	}
```
Comment (Emyrk) [inline agent/agentssh/portinspection_supported.go:24]: I see what you are going for here, but I think we can simplify.

```golang
// If both ipv4 and ipv6 throw an error, return an error.
if (err != nil && err6 != nil) {
  return "", xerrors.Errorf("inspect port %d: %w", port, errors.Join(err, err6))
}
```

If only 1 throws an error, then the function still runs with the other just fine. We only care if they both fail.

If 1 fails and the other returns nothing, is that not just an accurate reading of nothing on that port?

Comment (code-asher) [reply agent/agentssh/portinspection_supported.go:24]: Or maybe I should check the address in the actual callback.

Comment (Emyrk) [reply agent/agentssh/portinspection_supported.go:24]: Ohh I see. Let's just comment this appropriately then. And lets name it `err4` and `err6`

```golang
// In the common case, we want to check ipv4 listening addresses. 
// If this fails, we should return an error. 
// We also need to check ipv6.
// The assumption is, if we have an err4, and 0 ipv6 addresses listed, then
// we are interested in the err4 (and vice versa).
// So return both errors (at least 1 is non-nil) if the other list
// is empty.
if (err4 != nil && len(tabs6) == 0) || (err6 != nil && len(tabs) == 0) {
```

**Review by Emyrk — APPROVED** [review]

**Pull request merged**